flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -627,7 +627,7 @@ private void configureCheckpointing() {
 					CheckpointRetentionPolicy.RETAIN_ON_FAILURE :
 					CheckpointRetentionPolicy.RETAIN_ON_CANCELLATION;
 		} else {
-			retentionAfterTermination = CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION;
+			retentionAfterTermination = CheckpointRetentionPolicy.RETAIN_ON_FAILURE;
```

Comment:
Why do you want to change the default value here, instead of set an external checkpoint cleanup?

Reply:
Hundreds of flink production applications, most applications do not have set an external checkpoint cleanup. After the application fails, it cannot be recovered according to the checkpoint, and the recovery also brings a lot of trouble, which brings huge losses to the production service. Therefore, the default is RETAIN_ON_FAILURE is more reasonable.